Multiple sharedInformer creation: NullPointerException while running

While migrating a project that uses many watchers for different object types to shared informers, I am facing the following error after informers get started:
```
2021-04-09 08:48:43 WARNING io.fabric8.kubernetes.client.informers.cache.Controller run Reflector list-watching job exiting because the thread-pool is shutting down
java.util.concurrent.RejectedExecutionException: Error while starting ReflectorRunnable watch
	at io.fabric8.kubernetes.client.informers.cache.Reflector.listAndWatch(Reflector.java:87)
	at io.fabric8.kubernetes.client.informers.cache.Controller.run(Controller.java:109)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.NullPointerException
```
I tried to reuse the `sharedInformer` or have different ones, but in both cases, I see this exception happening around 60 times for a total of 120 created informers.

--- a/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/cache/Cache.java
+++ b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/cache/Cache.java
@@ -390,6 +390,9 @@ public class Cache<T> implements Indexer<T> {
    */
   public static String metaNamespaceKeyFunc(Object obj) {
     try {
+      if( obj == null ) {
+        return "";
+      }
       ObjectMeta metadata;
       if(obj instanceof String) {
         return (String) obj;

--- a/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/ReflectUtils.java
+++ b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/ReflectUtils.java
@@ -27,6 +27,9 @@ import java.util.List;
 public class ReflectUtils {
 
   public static ObjectMeta objectMetadata(Object obj) throws ReflectiveOperationException {
+    if( obj == null ) {
+      return null;
+    }
     if (obj instanceof HasMetadata) {
       return ((HasMetadata) obj).getMetadata();
     }
@@ -39,6 +42,9 @@ public class ReflectUtils {
   }
 
   public static String namespace(Object obj) throws ReflectiveOperationException {
+    if( obj == null ) {
+      return "";
+    }
     return objectMetadata(obj).getNamespace();
   }
 

--- a/kubernetes-client/src/test/java/io/fabric8/kubernetes/client/informers/cache/CacheTest.java
+++ b/kubernetes-client/src/test/java/io/fabric8/kubernetes/client/informers/cache/CacheTest.java
@@ -87,6 +87,7 @@ class CacheTest {
     Pod testPodObj = new PodBuilder().withNewMetadata().withName("test-pod4").withNamespace("default").endMetadata().build();
 
     cache.add(testPodObj);
+    assertEquals("", Cache.metaNamespaceKeyFunc(null));
     assertEquals("default/test-pod4", Cache.metaNamespaceKeyFunc(testPodObj));
     assertEquals("default/test-pod4", Cache.namespaceKeyFunc("default", "test-pod4"));
   }

--- a/kubernetes-client/src/test/java/io/fabric8/kubernetes/client/utils/ReflectUtilsTest.java
+++ b/kubernetes-client/src/test/java/io/fabric8/kubernetes/client/utils/ReflectUtilsTest.java
@@ -15,8 +15,10 @@
  */
 package io.fabric8.kubernetes.client.utils;
 
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
 import static org.junit.jupiter.api.Assertions.assertSame;
 
+import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
 import org.junit.jupiter.api.Test;
 
 import io.fabric8.kubernetes.api.model.HasMetadata;
@@ -91,6 +93,42 @@ public class ReflectUtilsTest {
     Baz baz = new Baz();
     baz.setMetadata(meta);
     assertSame(meta, ReflectUtils.objectMetadata(baz));
+  }
+
+  @Test
+  void testObjectMetadataReturnsNullOnNull() throws ReflectiveOperationException {
+    // Given
+    Object input = null;
+
+    // When
+    ObjectMeta result = ReflectUtils.objectMetadata(input);
+
+    // Then
+    assertThat(result).isNull();
+  }
+
+  @Test
+  void testNamespaceReturnsValidNamespace() throws ReflectiveOperationException {
+    // Given
+    Object input = new ConfigMapBuilder().withNewMetadata().withNamespace("ns1").endMetadata().build();
+
+    // When
+    String result = ReflectUtils.namespace(input);
+
+    // Then
+    assertThat(result).isNotNull().isEqualTo("ns1");
+  }
+
+  @Test
+  void testNamespaceReturnsBlankNamespace() throws ReflectiveOperationException {
+    // Given
+    Object input = null;
+
+    // When
+    String result = ReflectUtils.namespace(input);
+
+    // Then
+    assertThat(result).isNotNull().isBlank();
   }
 
 }

